salt/modules/ldap2.py

```diff
@@ -0,0 +1,423 @@
+# -*- coding: utf-8 -*-
+'''
+Query and modify an LDAP database (alternative interface)
+=========================================================
+
+.. versionadded:: Boron
+
+This is an alternative to the ``ldap`` interface provided by the
+:py:mod:`ldapmod <salt.modules.ldapmod>` execution module.
+
+:depends: - ``ldap`` Python module
+'''
+
+from __future__ import absolute_import
+
+try:
+    import ldap
+    import ldap.ldapobject
+    import ldap.modlist
+    import ldap.sasl
+    HAS_LDAP = True
+except ImportError:
+    HAS_LDAP = False
+import logging
+import salt.ext.six as six
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''Only load this module if the Python ldap module is present'''
+    return HAS_LDAP
+
+
+def _bind(l, bind=None):
+    '''Generic bind helper.'''
+    if bind is None:
+        return
+    method = bind.get('method', 'simple')
+    if method is None:
+        return
+    try:
+        bindfn = globals()['_bind_' + method]
```

Comment:
@cachedout: I wrote it this way to make it easy to add additional bind methods.  I don't really see any other bind methods being added, so I can do away with this inspection.